bhttp.go

```diff
@@ -568,7 +568,7 @@ func UnmarshalBinaryResponse(data []byte) (*http.Response, error) {
 	// Trailer
 	trailerFields := new(fieldList)
 	encodedFieldData, err = readVarintSlice(b)
-	if err != nil {
+	if err != io.EOF {
```

Comment:
This would return `nil` if the error was `nil`, right? The request trailer parser skips the trailer decoding if the trailer was omitted. We should probably do the same thing here. (Is this needed urgently?)

Reply:
Hahah, right, we still need to check for `nil` as well.

That was preventing response from Cloudflare's DOOH server from being decoded.

Reply:
Fixed, thanks :)